researchapp/services/oauth.py

```diff
@@ -35,7 +32,7 @@ def refresh(token):
     client_auth = requests.auth.HTTPBasicAuth('app-demo',
                                               'demo-secret-s4s')
 
-    response = requests.post(TOKEN_URL,
+    response = requests.post(provider['oauth_url'] + '/token',
```

Comment:
We shouldn't assume that the token URL can be constructed this way - the right way to learn it is via /metadata endpoint from the FHIR server